File: RokidSDKDemoJava/app/src/main/java/com/rokid/mobile/sdk/demo/java/skill/SkillAlarmAddActivity.java
package com.rokid.mobile.sdk.demo.java.skill;

import android.content.Context;
import android.support.annotation.Nullable;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.RelativeLayout;
import android.widget.TextView;
import android.widget.TimePicker;

import com.rokid.mobile.lib.base.util.Logger;
import com.rokid.mobile.lib.xbase.channel.IChannelPublishCallback;
import com.rokid.mobile.sdk.RokidMobileSDK;
import com.rokid.mobile.sdk.annotation.SDKRepeatType;
import com.rokid.mobile.sdk.bean.SDKAlarm;
import com.rokid.mobile.sdk.demo.java.R;
import com.rokid.mobile.sdk.demo.java.base.BaseActivity;
import com.rokid.mobile.sdk.demo.java.view.IconTextView;

import java.util.HashMap;
import java.util.Map;

import static com.rokid.mobile.sdk.demo.java.DemoApplication.getContext;

public class SkillAlarmAddActivity extends BaseActivity {

    public final static String DEVICE_ID = "device_id";

    private RelativeLayout titlebarLl;

    private IconTextView backIcon;

    private TextView titleTxt;

    private DatePicker datePicker;

    private TimePicker timePicker;

    private Button addBtn;

    private String deviceId;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_skill_alarm_add;
    }

    @Override
    protected void initVariables(@Nullable Bundle savedInstanceState) {
        titlebarLl = (RelativeLayout) findViewById(R.id.alarm_add_titlebar);
        backIcon = titlebarLl.findViewById(R.id.base_titlebar_left);
        titleTxt = titlebarLl.findViewById(R.id.base_titlebar_title);

        datePicker = (DatePicker) findViewById(R.id.alarm_date);
        timePicker = (TimePicker) findViewById(R.id.alarm_time);
        addBtn = (Button) findViewById(R.id.alarm_add_btn);

        titleTxt.setText("添加闹钟");
        initTimerPicker();

        deviceId = getIntent().getStringExtra(DEVICE_ID);
    }

    @Override
    protected void initListeners() {
        backIcon.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        addBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                addAlarm();
            }
        });
    }

    private void addAlarm() {
        Map<String, String> map = new HashMap<>();
        map.put("TestKey", "TestValue");

        SDKAlarm alarm = SDKAlarm.builder()
                .year(datePicker.getYear())
                .month(datePicker.getMonth() + 1)
                .day(datePicker.getDayOfMonth())
                .hour(timePicker.getCurrentHour())
                .minute(timePicker.getCurrentMinute())
                .repeatType(SDKRepeatType.EVERY_MONDAY)
                .ext(map)
                .build();

        RokidMobileSDK.skill.alarm().add(deviceId, alarm, new IChannelPublishCallback() {
            @Override
            public void onSucceed() {
                showToastShort("添加闹钟成功");
                finish();
            }

            @Override
            public void onFailed() {
                showToastShort("添加闹钟失败");

            }
        });
    }

    private void initTimerPicker() {
        if (get24HourMode(getContext())) {
            Logger.d("system is 24 hour");
            timePicker.setIs24HourView(true);
        } else {
            Logger.d("system is 12 hour");
            timePicker.setIs24HourView(false);
        }
    }

    private boolean get24HourMode(final Context context) {
        return android.text.format.DateFormat.is24HourFormat(context);
    }
}

File: RokidSDKDemoKotlin/app/src/main/kotlin/com/rokid/mobile/sdk/demo/skill/SkillAlarmAddActivity.kt
package com.rokid.mobile.sdk.demo.skill

import android.annotation.SuppressLint
import com.rokid.mobile.lib.xbase.channel.IChannelPublishCallback
import com.rokid.mobile.sdk.RokidMobileSDK
import com.rokid.mobile.sdk.annotation.SDKRepeatType
import com.rokid.mobile.sdk.bean.SDKAlarm
import com.rokid.mobile.sdk.callback.SDKOperateAlarmCallback
import com.rokid.mobile.sdk.demo.R
import com.rokid.mobile.sdk.demo.base.BaseActivity
import kotlinx.android.synthetic.main.base_titlebar.*
import kotlinx.android.synthetic.main.skill_activity_alarm_add.*

/**
 * Description: TODO
 * Author: Shper
 * Version: V0.1 2018/3/2
 */
class SkillAlarmAddActivity : BaseActivity() {

    companion object {
        val DEVICE_ID: String = "device_id"
    }

    var deviceId: String? = ""

    override fun layoutId(): Int = R.layout.skill_activity_alarm_add

    override fun initVariable() {
        super.initVariable()
        deviceId = intent.getStringExtra(DEVICE_ID)
    }

    override fun initViews() {
        base_titlebar_title.text = "添加闹钟"
        alarm_time.setIs24HourView(true)
    }

    @SuppressLint("NewApi")
    override fun initListeners() {
        base_titlebar_left.setOnClickListener { finish() }

        alarm_add_btn.setOnClickListener {
            val sdkAlarm = SDKAlarm().apply {
                year = alarm_date.year
                month = (alarm_date.month + 1)
                day = alarm_date.dayOfMonth
                hour = alarm_time.hour
                minute = alarm_time.minute
                repeatType = SDKRepeatType.ONCE
                ext = HashMap<String, String>().apply {
                    put("TestKey", "TestValue")
                }
            }

            //添加云闹钟
//            RokidMobileSDK.skill.cloudAlarm().addAlarm(deviceId, sdkAlarm, object : SDKOperateAlarmCallback {
//                override fun onSucceed() {
//                    toast("添加成功")
//                    finish()
//                }
//
//                override fun onFailed(p0: String?, p1: String?) {
//                    toast("添加失败")
//                }
//
//            })

            //添加本地闹钟
            RokidMobileSDK.skill.alarm().add(deviceId!!, sdkAlarm, object : IChannelPublishCallback {
                override fun onSucceed() {
                    toast("添加成功")
                    finish()
                }

                override fun onFailed() {
                    toast("添加失败")
                }

            })

        }
    }

}